### DISCUSSION TITLE
How to change decoder on fly with EncodedAudioStream?

### DISCUSSION THREAD

**Discussion by BeijingUncle** [pschatzmann/arduino-audio-tools#646] [Q&A] [answered]
Hi, Phil:

I'm trying to play different format audio file with EncodedAudioStream, 

Here is the declare of my sketch:

A2DPStream a2dp; 
MP3DecoderHelix mp3_decoder;
AACDecoderHelix aac_decoder;
WAVDecoder wav_decoder;
FLACDecoder flac_decoder;
EncodedAudioStream out2dec(&a2dp, &mp3_decoder);
StreamCopy copier(out2dec, audioFile);

but it seems there is not a method to change decoder on fly, How can I do that?

**Comment by pschatzmann** [comment, marked as answer]
I think the best is to declare a separate EncodedAudioStream for each decoder and switch the output dynamically in the copier. For details I suggest to have a look at the class documentation of StreamCopy
